frontend/src/user/store/user.reducers.ts

```diff
@@ -0,0 +1,55 @@
+import * as userState from './user.states';
+import * as userActions from './user.actions';
+import { createSelector, createFeatureSelector } from '@ngrx/store';
+
+export const initialState: userState.State = {
+    errorMessage: null,
+    sending: null,
+    sent: null
+};
+
+export const userFeature = createFeatureSelector<userState.State>('current');
+
+
+export const Errors = createSelector(
+    userFeature,
+    (state: userState.State) => state.errorMessage
+);
+
+export const SendSuccess = createSelector(
+    userFeature,
+    (state: userState.State) => state.sent
+);
+
+export function userReducer (
+    state: userState.State = initialState,
+    action: userActions.All
+    ): userState.State {
+        switch (action.type) {
+            case userActions.currentTypes.SEND:
+            return {
+                errorMessage: null,
```

Comment:
nie musisz tutaj używać errorMessage, initial state już ma nulla.
Dwa razy ci przypisze nulla

Reply:
https://imgur.com/a/vmkPbqi
Chyba jednak musi być, chyba, że czegoś nie rozumiem. Nie moge mu dać returna bez danego pola z initState, a jak dam ...state to przepisuje poprzedni stan i error będzie all the time w storze